rpc/client.go

```diff
@@ -14,205 +14,32 @@
-		if err != nil {
-			err = errors.Wrapf(err, "init yamux client to %s failed", nodeID)
-			return
-		}
-		conn, err = sess.OpenStream()
-		if err != nil {
-			err = errors.Wrapf(err, "open new session to %s failed", nodeID)
-		}
-		return
-	}
-	//log.WithField("poolSize", pool.Len()).Debug("session pool size")
-	conn, err = pool.Get(nodeID)
-	return
-}
-
-// dialToNode connects to the node with nodeID.
-func dialToNode(nodeID proto.NodeID) (conn net.Conn, err error) {
-	return dialToNodeEx(nodeID, false)
-}
-
-// dialToNodeEx connects to the node with nodeID.
-func dialToNodeEx(nodeID proto.NodeID, isAnonymous bool) (conn net.Conn, err error) {
-	var rawNodeID = nodeID.ToRawNodeID()
-	/*
-		As a common practice of PKI, we should add some randomness to the ECDHed pre-master-key
-		we did that at the [ETLS](../crypto/etls) layer with a non-deterministic authenticated
-		encryption input vector(random IV).
-
-		To understand that and func "rpc.GetSharedSecretWith"
-		Please refer to:
-			- RFC 5297: Synthetic Initialization Vector (SIV) Authenticated Encryption
-				Using the Advanced Encryption Standard (AES)
-			- RFC 5246: The Transport Layer Security (TLS) Protocol Version 1.2
-		Useful links:
-			- https://tools.ietf.org/html/rfc5297#section-3
-			- https://tools.ietf.org/html/rfc5246#section-5
-			- https://www.cryptologie.net/article/340/tls-pre-master-secrets-and-master-secrets/
-	*/
-	symmetricKey, err := GetSharedSecretWith(rawNodeID, isAnonymous)
-	if err != nil {
-		return
-	}
-
-	nodeAddr, err := GetNodeAddr(rawNodeID)
-	if err != nil {
-		err = errors.Wrapf(err, "resolve %s failed", rawNodeID.String())
-		return
-	}
-
-	cipher := etls.NewCipher(symmetricKey)
-	conn, err = dial("tcp", nodeAddr, rawNodeID, cipher, isAnonymous)
-	if err != nil {
-		err = errors.Wrapf(err, "connect %s %s failed", rawNodeID.String(), nodeAddr)
-		return
-	}
-
-	return
-}
-
-// NewClient returns a RPC client.
-func NewClient() *Client {
-	return &Client{}
-}
-
-// initClient initializes client with connection to given addr.
-func initClient(addr string) (client *Client, err error) {
-	conn, err := net.Dial("tcp", addr)
-	if err != nil {
-		return nil, err
-	}
-	return InitClientConn(conn)
-}
-
-// InitClientConn initializes client with connection to given addr.
-func InitClientConn(conn net.Conn) (client *Client, err error) {
-	client = NewClient()
-	var muxConn *mux.Stream
-	muxConn, ok := conn.(*mux.Stream)
-	if !ok {
-		var sess *mux.Session
-		sess, err = mux.Client(conn, MuxConfig)
-		if err != nil {
-			err = errors.Wrap(err, "init mux client failed")
-			return
-		}
-
-		muxConn, err = sess.OpenStream()
-		if err != nil {
-			err = errors.Wrap(err, "open stream failed")
-			return
-		}
-	}
-	client.Conn = muxConn
-	client.Client = rpc.NewClientWithCodec(utils.GetMsgPackClientCodec(muxConn))
-	client.RemoteAddr = conn.RemoteAddr().String()
-
-	return client, nil
+// Client defines the extend method to set client last error.
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported type LastErrSetter should be of the form "LastErrSetter ..." (with optional leading article)